pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
+	// Required: The IP of this endpoint.
+	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
```

Comment:
This is tough, because it puts the onus on people to denormalize DNS.  Ultimately endpoints aren't resolved, it's still the responsibility of the consumer (proxy, headless service) to resolve them.

Ie I might want to make an external saas connection have endpoints "myservice.internal.east1.myco.com" but I probably won't bother using a service if I have to keep that in sync with a DNS record


> On Feb 13, 2015, at 10:07 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > @@ -738,7 +771,35 @@ type Endpoints struct {
> >  	TypeMeta   `json:",inline"`
> >  	ObjectMeta `json:"metadata,omitempty"`
> >  
> > -	Endpoints []string `json:"endpoints,omitempty"`
> > +	Endpoints []Endpoint `json:"endpoints,omitempty"`
> > +}
> > +
> > +type Endpoint struct {
> > +	// Required: The IP of this endpoint.
> > +	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
> Did we permit hostname before? For Endpoint, in particular, I think it's best if it's an address, to avoid dependencies on DNS.
> 
> We should at least allow v6 as well as v4.
> 
> —
> Reply to this email directly or view it on GitHub.
>